airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/profiles.py

```diff
@@ -0,0 +1,73 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from typing import Any, Iterable, List, Mapping
+
+import requests
+from airbyte_cdk.models import SyncMode
+from source_amazon_ads.schemas import Profile
+from source_amazon_ads.streams.common import AmazonAdsStream
+
+
+class Profiles(AmazonAdsStream):
+    """
+    This stream corresponds to Amazon Advertising API - Profiles
+    https://advertising.amazon.com/API/docs/en-us/reference/2/profiles#/Profiles
+    """
+
+    primary_key = "profileId"
+    model = Profile
+
+    def path(self, **kvargs) -> str:
+        return "v2/profiles"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        for record in super().parse_response(response, **kwargs):
+            profile_id_obj = self.model.parse_obj(record)
+            # Populate self._profiles list with profiles objects to not make
+            # unnecessary API calls.
+            self._profiles.append(profile_id_obj)
```

Comment:
is this needed if `profiles` is now being passed as explicit input to other classes?

Reply:
Yes, Profiles stream still can be read twice, this is cached profiles response.